react-app/src/components/PortfolioScreen/PortfolioScreenAPI.ts

```diff
@@ -104,18 +104,21 @@ export const usePortfolioQuery = (): {
         (delegation) => delegation.delegation.validatorAddress
       );
 
-      const rewards = await distribution.getDelegationRewardsByValidators(
-        address,
-        validatorAddresses
-      );
+      const [rewards, validators, expectedReturns] = await Promise.all([
+        distribution.getDelegationRewardsByValidators(
+          address,
+          validatorAddresses
+        ),
+        stakingAPI.getValidators(validatorAddresses),
+        distribution.getBatchValidatorExpectedReturn(validatorAddresses),
```

Comment:
I think the expectedReturn calculation can be moved to screenAPI level as we are getting 2n validators here. Let's open issue for improvement.